lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -156,8 +387,9 @@ createSpecializedType(CanSILFunctionType SubstFTy, SILModule &M) const {
       SILType SILParamTy = SILType::getPrimitiveObjectType(PI.getType());
       // Indirect parameters are passed as owned, so we also need to pass the
       // direct parameter as owned (except it's a trivial type).
-      auto C = (SILParamTy.isTrivial(M) ? ParameterConvention::Direct_Unowned :
-                ParameterConvention::Direct_Owned);
+      auto C = ((!SILParamTy.hasTypeParameter() && SILParamTy.isTrivial(M))
```

Comment:
Ah, you mean `mapTypeIntoContext(SILType)`? This one is defined in the SILFunction class. But we don't have a SILFunction object yet. We are actually creating a type which is to be used by SILFunction::create.... I need to think how to get around this. It would be unfortunate to duplicate the code.